src/cocotb/__init__.py

```diff
@@ -306,8 +307,9 @@ def _sim_event(message):
     # We simply return here as the simulator will exit
     # so no cleanup is needed
     msg = f"Failing test at simulator request before test run completion: {message}"
-    scheduler.log.error(msg)
-    scheduler._finish_scheduler(SimFailure(msg))
+    log.error(msg)
```

Comment:
When parameterize would get errors, a subsequent error would arise here from scheduler being None

Reply:
This is being fixed in #3508. Lets leave this change for that PR,